data_catalog_backend/routes/admin/resource_routes.py

```diff
@@ -45,3 +46,25 @@ async def add_resource(
         raise HTTPException(
             status_code=500, detail=f"Error creating resource: {str(e)}"
         )
+
+
+@router.delete(
+    "/{resource_id}",
+    description="Delete a resource from the metadata store",
+    tags=["admin"],
+    response_model=ResourceResponse,
+    response_model_exclude_none=True,
+)
+async def delete_resource(
+    resource_id: uuid.UUID,
+    current_user: Annotated[User, Depends(authenticate_user)],
+    service: ResourceService = Depends(get_resource_service),
+) -> ResourceResponse:
+    try:
+        deleted_resource = service.delete_resource(resource_id, current_user)
+        return ResourceResponse.model_validate(deleted_resource)
+    except Exception as e:
```

Comment:
This error-handling is sloppy.
You are even omitting the check for if the resource exists in the service layer. This needs to be more specific, and more idiomatic.